db/migrate/20180426135126_add_new_fields_to_attendance_events.rb

```diff
@@ -0,0 +1,13 @@
+class AddNewFieldsToAttendanceEvents < ActiveRecord::Migration[5.1]
+  def change
+    add_column :absences, :dismissed, :boolean
+    add_column :absences, :excused, :boolean
+    add_column :absences, :reason, :string
+    add_column :absences, :comment, :string
+
+    add_column :tardies, :dismissed, :boolean
+    add_column :tardies, :excused, :boolean
+    add_column :tardies, :reason, :string
+    add_column :tardies, :comment, :string
```

Comment:
I'm assuming that by default these columns will be serialized and sent down to the UI when we ship this, is that your expectation too?  I don't think it matters a whole lot, but could increase the size a bit for the absences and tardies dashboard if they aren't already strict about limiting the fields they send down to the UI.

Reply:
Yes, the columns will be serialized and sent to the UI for the Absences and Tardies dashboards, though both those queries are scoped to absences and tardies for the past year. Will also add more data to the UI for the Student Profile. Won't affect the School Overview page since that uses `most_recent_school_year_absences_count`. The Home Page for educators doesn't sent serialized absences to the UI. 

I think we should keep an eye out for how much of an effect this has on performance for the Absences/Tardies page and the student profile.

Reply:
Cool, that makes sense!

Looking at the shape of the data here, one thing I notice is that this adds several string columns, and based on your awesome digging up above, these are present for ~30% of the absence events.  For the HS absences page, right now we're sending down ~10k absence events.  So if we take 30% of those and double or triple the size of the event object we send down, it seems like there's a good chance this would have a noticeable effect.  It'd be a bummer to ship this and then have the awesome performance improvement gains that you made last week regress.

So two suggestions might be: ship this and then measure the impact on the payload size for the absences page and page load time, or update `SchoolsController#absence_dashboard_data` first to whitelist only the fields we need in the UI.  Then this PR is less likely to have an impact on the absences view.

Reply:
I like the whitelisting strategy! I think that also gives us the opportunity to ship some exploratory features around this dark, where we can surface fields like `reason` and `excused` and get an understanding of how educators are interpreting what they see.

Reply:
@alexsoble Sounds good!